contracts/rewards/src/contract/execute.rs

```diff
@@ -59,7 +61,10 @@ pub(crate) fn distribute_rewards(
     epoch_process_limit: Option<u64>,
 ) -> Result<HashMap<Addr, Uint128>, ContractError> {
     let epoch_process_limit = epoch_process_limit.unwrap_or(DEFAULT_EPOCHS_TO_PROCESS);
-    let cur_epoch = Epoch::current(&state::load_params(storage), cur_block_height)?;
+    let cur_epoch = Epoch::current(
+        &state::load_rewards_pool_params(storage, pool_id.clone())?,
+        cur_block_height,
+    )?;
```

Comment:
create a `state::current_epoch(storage, pool_id, block_height)` function to wrap this?